packages/core/src/plugins/types.ts

```diff
@@ -0,0 +1,3 @@
+export interface PluginContext {
```

Comment:
Nit: can we place this in an existing file instead of creating a new file? Maybe `Plugin.js`?

Reply:
Just left a [comment](https://github.com/MarkBind/markbind/issues/2114#issuecomment-1413805848) on #2114 related to this - my concern was that putting types in an existing file could create cyclic dependency issues. We shouldn't have cyclic dependencies in this case for plugins, but if we choose to standardise how the types are handled, a `types.ts` may be a solution. What do you think? For now I could leave it in Plugin.js and see how it goes?

Reply:
Perhaps can also consider using the `d.ts` extension? Somehow `types.ts` feels like an anti-pattern to me.

Reply:
I think as long as we standardize how we organize the types, it _should_ be fine. I would prefer something more descriptive than `types.ts` though. 

> For now I could leave it in Plugin.js and see how it goes?

Yup! I think you can do this for now. Let's finish up with migration first since it is higher priority and we can come back later to reorganize the types.

Reply:
Speaking of Plugin.js - should we put this PR on hold until plugin.js is migrated, since the types will be defined there?

Reply:
I think merging this in first is fine, can refactor it later if needed.